src/scripts/Add-Scope.ps1

```diff
@@ -0,0 +1,241 @@
+          "type": "{3}",
+          "timeframe": "{4}",
+          "dataSet": {
+            "granularity": "Daily"
+          }
+        }
+      }
+    }'
+
+    $payload = $payload.Replace("{0}", $ApiParams.DestinationResourceId)
+    $payload = $payload.Replace("{1}", $ApiParams.DestinationContainer)
+    $payload = $payload.Replace("{2}", $ApiParams.DestinationRootFolderPath)
+    $payload = $payload.Replace("{3}", $ApiParams.DefinitionType)
+    $payload = $payload.Replace("{4}", $ApiParams.DefinitionTimeframe)
+    $payload = $payload.Replace("{5}", $ApiParams.PartitionData)
+    $payload = $payload.Replace("{6}", $ApiParams.RecurrencePeriodFrom)
+    $payload = $payload.Replace("{9}", $ApiParams.RecurrencePeriodTo)
+    $payload = $payload.Replace("{7}", $ApiParams.ScheduleRecurrence)
+    $payload = $payload.Replace("{8}", $ApiParams.ScheduleStatus)
+  }
+  
+  $apiResult = Invoke-AzRestMethod -Uri $uri -Method PUT -Payload $payload
+  if ($apiResult.StatusCode -ne "201") {
+    $apiResult
+    Throw "Cost Management API call failed"
+  }
+}
+
+function Set-CostManagementExport {
+  param (
+    $ExportParams,
+    [bool]$Start = $false
+  )
+
+  Write-DebugInfo $ExportParams
+  Set-CostManagementApi -ApiParams $ExportParams
+  if ($Start) {
+    Start-Sleep -Seconds $SleepInterval
+    Invoke-AzCostManagementExecuteExport -ExportName $ExportParams.Name -Scope $ExportParams.Scope
+    Start-Sleep -Seconds $SleepInterval
+    $currentStatus = $null
+    $currentStatus = (Get-AzCostManagementExport -Name $ExportParams.Name -Scope $ExportParams.Scope -Expand runHistory).RunHistory.Value[0].Status
+    [int]$loop = 0
+    while ($currentStatus -eq "InProgress") {
+      if ($loop -ge $TimeOutMinutes) {
+        $currentStatus = "TimedOut"
+      }
+      else {
+        Start-Sleep -Seconds 60
+        $loop++
+        $currentStatus = (Get-AzCostManagementExport -Name $ExportParams.Name -Scope $ExportParams.Scope -Expand runHistory).RunHistory.Value[0].Status
+        Write-Host ("{0}    {1}    {2}" -f (get-date), $ExportParams.Name, $currentStatus)
+      }
+    }
+
+    Write-Host ("{0}    {1}    {2}" -f (get-date), $ExportParams.Name, $currentStatus)
+  }
+}
+
+if ($Future) {
+  Write-Host ("{0}    {1}" -f (get-date), "Set Recurring Exports")
+  $today = Get-Date
+  $nextMonth = $today.AddDays(-$today.Day + 5).AddMonths(1)
+  [string]$dateFrom = "{0}-{1}-{2}T10:00:00Z" -f $nextMonth.Year, $nextMonth.Month, $nextMonth.Day
+
+  # Last Billing Month
+  $exportName = "{0}-{1}-c-{2}" -f $Metric, $Scope.Split('/')[$Scope.Split('/').Length - 1], $HubName
+  $Params = @{
+    Name                      = $exportName.substring(0, [System.Math]::Min(63, $exportName.Length))
+    DefinitionType            = $Metric
+    DataSetGranularity        = 'Daily'
+    Scope                     = $Scope
+    DestinationResourceId     = $StorageAccountId
+    DestinationContainer      = $ContainerName
+    DefinitionTimeframe       = 'TheLastBillingMonth'
```

Comment:
Can we use `LastMonth` (or whatever the calendar-based month is)? I'd rather avoid the issues that come in when we support non-calendar billing periods.

Reply:
Not an option for recurring exports.  Options are The LastBillingMonth, BillingMonthToDate and Custom